File: src/com/kingmang/lazurite/parser/ast/Argument.java
package com.kingmang.lazurite.parser.ast;

public final class Argument {

    private final String name;
    private final Expression valueExpr;

    public Argument(String name) {
        this(name, null);
    }

    public Argument(String name, Expression valueExpr) {
        this.name = name;
        this.valueExpr = valueExpr;
    }

    public String getName() {
        return name;
    }

    public Expression getValueExpr() {
        return valueExpr;
    }

    @Override
    public String toString() {
        return name + (valueExpr == null ? "" : " = " + valueExpr);
    }
}

File: src/main/java/com/kingmang/lazurite/parser/ast/Argument.kt
package com.kingmang.lazurite.parser.ast

import com.kingmang.lazurite.parser.ast.expressions.Expression

data class Argument(
    val name: String,
    val valueExpr: Expression? = null
) {
    override fun toString(): String =
        "${this.name}${this.valueExpr?.let { " = $it" }.orEmpty()}"
}
